draft-ietf-acme-acme.md

```diff
@@ -509,7 +509,8 @@ any signed request from the client to carry such a nonce.
 An ACME server provides nonces to clients using the HTTP Replay-Nonce header field,
 as specified in {{replay-nonce}} below.  The server MUST include a Replay-Nonce
 header field in every successful response to a POST request and SHOULD provide
-it in error responses as well.
+it in error responses as well.  Servers SHOULD use globally scoped nonces, so that
```

Comment:
There's an interop concern here, in that clients need to know how they can use nonces.  I would probably make the slightly stronger argument that we should forbid servers from doing things that would break the obvious ways for clients to handle nonces (i.e., in one pool per client).

For example, a server could have a scheme where (1) a `newNonce`-issued nonce is good for talking to any resource, but (2) nonces issued from other resources can only be used with resources of the same class.  That would cause a lot of fail/newNonce/retry loops for a client doing the obvious thing.

TBH, incorporating client IP seems kind of on the edge, as it's not crazy to imagine that a client could change addresses during the course of an ACME issuance flow.

How about something like this:

> Servers SHOULD NOT scope nonces more finely than per-account.  In particular, servers MUST ensure that a nonce issued to a client by the `newNonce` resource can be used by the same client to perform any ACME request (within a reasonable period of time).

The latter part ensures that the fail/newNonce/retry loop works.

Reply:
> There's an interop concern here, in that clients need to know how they can use nonces. 

I don't think that's true. Clients that send a nonce to a server from the wrong scope will get back a nonce rejection error that itself has a new nonce to use that the server will accept subsequently. The spec already says that a client SHOULD retry in this case. Some imagined (e.g. not implemented, not discussed on list by anyone) server nonce implementations like per-resource scoping might provoke more retries from clients but that seems like a fair trade-off for a hypothetical design choice and not a concern that justifies changing the draft text.

> How about something like this:
>> Servers SHOULD NOT scope nonces more finely than per-account.

I don't think the text should have a "SHOULD NOT" added that directly contradicts all of the existing server implementations I'm aware of, and certainly the largest production ACME implementation and its ecosystem of compatible clients.

At this stage of the process I really think that every change should have an extremely strong justification. This change contradicts all implementation experience we have and the rationale for it (e.g. supporting servers implementing per-resource nonce schemes) does not seem strong enough to meet the bar for merging.